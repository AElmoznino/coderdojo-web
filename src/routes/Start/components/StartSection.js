// @flow

import React from 'react'
import styled from 'styled-components'
import ButtonLink from '../../../components/Button/ButtonLink'
import H1 from '../../../components/Typography/H1'
import Wrap from '../../../components/Grid/GridSection'

type StartSectionProps = {
  link: string,
  text: string,
  title: string,
}

const InnerWrap = styled.div`
  display: grid;
  grid-template-columns: 200px 1fr;
  grid-column-gap: ${({ theme }) => theme.sizes.medium};
`

const RightWrap = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
`

const AlignedButton = styled(ButtonLink)`
  align-self: flex-end;
`

const StartSection = ({ link, text, title }: StartSectionProps) => (
  <Wrap>
    <H1>{title}</H1>
    <InnerWrap>
      <img alt="placeholder alt" src="https://placeimg.com/200/200/tech" />
      <RightWrap>
        <p>{text}</p>
        <AlignedButton to={link}>Kom ig√•ng</AlignedButton>
      </RightWrap>
    </InnerWrap>
  </Wrap>
)

export default StartSection
