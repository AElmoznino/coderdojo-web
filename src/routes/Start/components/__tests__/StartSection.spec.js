import React from 'react'
import { shallow } from 'enzyme'
import StartSection from '../StartSection'

describe('components/StartSection', () => {
  let component

  beforeEach(() => {
    component = shallow(
      <StartSection link="testLink" text="testText" title="testTitle" />,
    )
  })

  it('renders StartSection', () => {
    expect(component).toMatchSnapshot()
  })
})
