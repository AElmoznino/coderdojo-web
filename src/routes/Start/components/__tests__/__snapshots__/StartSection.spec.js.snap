// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/StartSection renders StartSection 1`] = `
<styled.div>
  <styled.h1>
    testTitle
  </styled.h1>
  <styled.div>
    <img
      alt="placeholder alt"
      src="https://placeimg.com/200/200/tech"
    />
    <styled.div>
      <p>
        testText
      </p>
      <Styled(Link)
        to="testLink"
      >
        Kom ig√•ng
      </Styled(Link)>
    </styled.div>
  </styled.div>
</styled.div>
`;
